ios/FluentUI/Core/Theme/Tokens/TokenSet.swift

```diff
@@ -4,7 +4,37 @@
 //
 
 /// Template for all token sets, both global and alias. This ensures a unified return type for any given token set.
-protocol TokenSet {
-    associatedtype TokenType
-    var value: TokenType { get }
+public final class TokenSet<T: Hashable & CaseIterable, V> {
+
+    /// Allows us to index into this token set using square brackets.
+    ///
+    /// We can use square brackets to both read and write into this `TokenSet`. For example:
+    /// ```
+    /// let value = tokenSet[.primary]
+    /// tokenSet[.secondary] = newValue
+    /// ```
+    public subscript(token: T) -> V {
+        get {
+            if let value = valueOverrides?[token] {
+                return value
+            }
+            return defaultValues(token)
+        }
+        set(value) {
+            if valueOverrides == nil {
+                valueOverrides = [:]
+            }
+            valueOverrides?[token] = value
+        }
+    }
+
+    /// Initializes this token set with a callback to fetch its default values as needed.
+    ///
+    /// - Parameter defaultValues: The closure that can be used to lazily fetch default token values as needed.
```

Comment:
nit: Instead of using "lazily", how about clarifying that the closure provide static default values that will be provided in case no override is provided for that specific token?

Reply:
Simplified this comment, and added a lengthier description about how that is expected to work.